RecoLocalCalo/HGCalRecProducers/plugins/HGCalUncalibRecHitWorkerWeights.cc

```diff
@@ -4,9 +4,62 @@
 #include "FWCore/Framework/interface/Event.h"
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+template< typename DET >
+void configureIt(const edm::ParameterSet& conf, 
+                 HGCalUncalibRecHitRecWeightsAlgo<HGCDataFrame<DET,HGCSample>>& maker) {
+  constexpr char isSiFE[]           = "isSiFE";
+  constexpr char adcNbits[]         = "adcNbits";
+  constexpr char adcSaturation[]    = "adcSaturation";
+  constexpr char tdcNbits[]         = "tdcNbits";
+  constexpr char tdcSaturation[]    = "tdcSaturation";
+  constexpr char tdcOnset[]         = "tdcOnset";
+  constexpr char toaLSB_ns[]        = "toaLSB_ns";
+  
+  if( conf.exists(isSiFE) ) {
```

Comment:
This class needs to react different depending on it being for an Si readout vs. Scintillator readout.
Should probably be factored into two different classes in the future.